### ISSUE TITLE
Does not use authentication to download files from authenticated feed

### ISSUE THREAD

**Issue by paulgit** [gpodder/gpodder#227]
Using the CLI version of gPodder (gPodder 3.9.3 "Whacketts" (2016-12-22)) I added some authenticated podcast feeds (these are private feeds of my own). gPodder correctly authenticated and read the feeds, however when it came to downloading the files it did not send the authentication and failed, below is an example output:

> Downloading 2017-02-02 bhc                                               1486212982.473616 [gpodder.download] ERROR: HTT
> P 401 while downloading "2017-02-02 bhc": Unauthorized
> Traceback (most recent call last):
>   File "/home/tigluap/apps/gpodder/src/gpodder/download.py", line 772, in run
>     self.tempname, reporthook=self.status_updated)
>   File "/home/tigluap/apps/gpodder/src/gpodder/download.py", line 274, in retrieve_resume
>     fp = self.open(url, data)
>   File "/usr/lib/python2.7/urllib.py", line 208, in open
>     return getattr(self, name)(url)
>   File "/usr/lib/python2.7/urllib.py", line 451, in open_https
>     return self.http_error(url, fp, errcode, errmsg, headers)
>   File "/usr/lib/python2.7/urllib.py", line 376, in http_error
>     return self.http_error_default(url, fp, errcode, errmsg, headers)
>   File "/home/tigluap/apps/gpodder/src/gpodder/download.py", line 215, in http_error_default
>     raise gPodderDownloadHTTPError(url, errcode, errmsg)
> gPodderDownloadHTTPError

Other pod catchers seem to work fine.

**Comment by thp** [comment]
I've checked our implementation, and we should already do this, so yes this is probably a bug.
However, looking at the `urrlib` code, I see this:

```
    def http_error_401(self, url, fp, errcode, errmsg, headers, data=None):
        """Error 401 -- authentication required.
        This function supports Basic authentication only."""
        if not 'www-authenticate' in headers:
            URLopener.http_error_default(self, url, fp,
                                         errcode, errmsg, headers)
        stuff = headers['www-authenticate']
        import re
        match = re.match('[ \t]*([^ \t]+)[ \t]+realm="([^"]*)"', stuff)
        if not match:
            URLopener.http_error_default(self, url, fp,
                                         errcode, errmsg, headers)
        scheme, realm = match.groups()
        if scheme.lower() != 'basic':
            URLopener.http_error_default(self, url, fp,
                                         errcode, errmsg, headers)
        name = 'retry_' + self.type + '_basic_auth'
        if data is None:
            return getattr(self,name)(url, realm)
        else:
            return getattr(self,name)(url, realm, data)
```

The function [prompt_user_passwd()](https://github.com/gpodder/gpodder/blob/master/src/gpodder/download.py#L325) is actually there since 10 years ago https://github.com/gpodder/gpodder/commit/f84d9847804216b57ffb3d273c2459a8691e684a and it was further improved 8 years ago: https://github.com/gpodder/gpodder/commit/996b2bcda218c1bc9dff871dbb702fbecd6e5bd7

Judging from the above code snippet, could it be that the server doesn't send a `WWW-Authenticate` header? And/or that the authentication scheme is not `"basic"`?

**Comment by elelay** [comment]
This is weird: python behaviour changed in python [3.2](https://github.com/python/cpython/commit/80f1b059714aeb1c6fc9f6ce1173bc8a51af7dd9). Now it doesn't automatically call prompt_user_passwd() but raises an exception.
So, this is fixed in ad86c35b9a11d69088159b41c9d7c5647263ca37.
